### PR TITLE
MSI-787: "Disassociated Products" configurable grid doesn't show information correctly

### PR DESCRIPTION
**Pull request by RomaKis**
### Fixed Issues (if relevant)

1. https://github.com/magento-engcom/msi/issues/787

### CODE REVIEW

File: app/code/Magento/InventoryConfigurableProduct/Observer/ProcessSourceItemsObserver.php [hunk 1]
```diff
@@ -53,10 +61,17 @@ public function execute(EventObserver $observer)
...
-                    $this->processSourceItems($productData['quantity_per_source'], $sku);
+            foreach ($productsData as $key => $productData) {
+                if (isset($productData['quantity_per_source'])) {
+                    $quantityPerSource = is_array($productData['quantity_per_source'])
+                        ? $productData['quantity_per_source']
+                        : [];
+
+                    // get sku by child id, because child sku can be changed if product with such sku already exists.
+                    $childProductId = $product->getExtensionAttributes()->getConfigurableProductLinks()[$key];
+                    $childProduct = $this->productRepository->getById($childProductId);
```
Comment (naydav) [inline app/code/Magento/InventoryConfigurableProduct/Observer/ProcessSourceItemsObserver.php]: We don't need to load product
We have `\Magento\InventoryCatalog\Model\GetSkusByProductIdsInterface` for resolving sky by product id

File: app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+use Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix;
+use Magento\InventoryCatalog\Model\IsSingleSourceModeInterface;
+use Magento\InventoryConfigurableProduct\Model\GetQuantityInformationPerSource;
+
+/**
+ * Add value for field "quantityPerSource" for grid "Associated Products" and "Disassociated Products"
+ * on step "Summary".
+ */
+class AdaptMatrix
```
Comment (naydav) [inline app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php]: Generally, in MSI, we use Plugin prefix
Also, need to provide more clear name

Like, `AddQuantityPerSourceToVariationsMatrix`

File: app/code/Magento/InventoryConfigurableProduct/etc/adminhtml/di.xml [hunk 3]
```diff
@@ -31,4 +31,8 @@
             <argument name="multiSourceTemplate" xsi:type="string">Magento_InventoryConfigurableProduct::catalog/product/edit/attribute/steps/summary.phtml</argument>
         </arguments>
     </type>
+    <type name="Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix">
+        <plugin name="adapt_grids_on_summary_step_to_msi"
```
Comment (naydav) [inline app/code/Magento/InventoryConfigurableProduct/etc/adminhtml/di.xml]: `add_quantity_per_source_to_variations_matrix`

File: app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     * @return array
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterGetProductMatrix(
+        Matrix $subject,
+        $result
+    ): array {
+        if ($this->isSingleSourceMode->execute() === false) {
+            foreach ($result as $key => $variation) {
```
Comment (naydav) [inline app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php]: According to the base implementation, the result can be `null`
SO you will get PHP Warning `: Invalid argument supplied for foreach()`

File: app/code/Magento/InventoryConfigurableProduct/Observer/ProcessSourceItemsObserver.php (hunk 1 above)
Comment (RomaKis) [reply app/code/Magento/InventoryConfigurableProduct/Observer/ProcessSourceItemsObserver.php]: Fixed.

File: app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php (hunk 2 above)
Comment (RomaKis) [reply app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php]: Fixed.

File: app/code/Magento/InventoryConfigurableProduct/etc/adminhtml/di.xml (hunk 3 above)
Comment (RomaKis) [reply app/code/Magento/InventoryConfigurableProduct/etc/adminhtml/di.xml]: Fixed.

File: app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php (hunk 4 above)
Comment (RomaKis) [reply app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php]: Fixed.

**Pull request merged**